### DISCUSSION TITLE
How to fix apostrophes rendering as â€™ ?

### DISCUSSION THREAD

**Discussion by penyuan** [mkdocs/mkdocs#3329] [Q&A] [answered]
Hello, 

I'm running MKDocs 1.4.3 in a Python 3.11.2 virtual environment on Linux, using the MKDocs Material theme.

When I serve my site locally using `mkdocs serve`, I see that many symbols are rendered incorrectly. For example, `'` is rendered as `â€™`.

After some digging, looks like it has something to do with needing to use UTF-8 encoding instead of ISO-8859-1/Windows-1252 (see [this thread](https://stackoverflow.com/q/2477452/186904)).

OK, but how do I do that for MKDocs??? Any specific tips would be appreciated.

**Comment by pawamoy** [comment]
Did you try to re-encode your files in UTF8? For example in VSCode, you can click on the encoding in the bottom bar (already UTF8 here):

![encoding](https://github.com/mkdocs/mkdocs/assets/3999221/db2963b6-5480-4a86-9421-dfff5e7eb5a4)

Other editors have their own ways of re-encoding files.

You can also do that from the shell, for example:

```bash
for file in *; do
  iconv -t utf8 $file -o $file
done
```

(not tested)

**Reply by penyuan** [reply]
Thank you @pawamoy for the quick response.

Indeed I'm using VSCode, but the bottom bar already says "UTF-8". I also re-saved the file using your `iconv` command. But neither solution solved this problem...

Any other suggestions?

**Reply by pawamoy** [reply]
Not sure. Check your browser settings :shrug:
We'd need more info about your setup (ideally a link to your repo) to investigate further.

**Reply by penyuan** [reply]
Thanks @pawamoy!

I am running Firefox 116.0 and my colleague can reproduce this problem on their macOS system as well.

> We'd need more info about your setup (ideally a link to your repo) to investigate further.

Here is the repository that I'm working from: 

https://github.com/elixir-europe-training/ELIXIR-TrP-ContainersPython-CodeRep

**Comment by pawamoy** [comment]
The issue partly comes from mkdocs-video, which loads and redumps the HTML contents through LXML, see https://github.com/soulless-viewer/mkdocs-video/blob/master/mkdocs_video/plugin.py. Try disabling this plugin and you'll get better results.

For &mldr;, try using `&mldr;` instead.

For the other characters (like `Ž`), I'm not sure yet.

**Reply by penyuan** [reply]
Thanks @pawamoy this worked! But I'd like to know what you mean by adding `{{ super() }}` to the `site_meta` block as described [here](https://github.com/mkdocs/mkdocs/discussions/3329#discussioncomment-6618114).

**Comment by pawamoy** [comment]
OK found it. You override the `site_meta` block but forgot to include the parent content, so the `<meta charset="UTF-8">` tag was dropped.

Put this in your `site_meta` block: `{{ super() }}` (beginning or end, not important here).

**Reply by penyuan** [reply]
Thanks @pawamoy disabling the `mkdocs-video` plugin worked as you described [here](https://github.com/mkdocs/mkdocs/discussions/3329#discussioncomment-6617998)!

For the `site_meta` block, I wasn't the one who created it and don't really know how that works. Is it this file: `./overrides/main.html`? If so, I tried copy-pasting `{{ super() }}` into it, but it's not clear to me exactly where it should go. Does it go within `<script type="application/ld+json">`? Or within `{% block site_meta %}`? I tried pasting it into a few difference places but none solved the problem...

**Reply by pawamoy** [reply, marked as answer]
See https://squidfunk.github.io/mkdocs-material/customization/#overriding-blocks.

```console
% cat overrides/main.html 
{% extends "base.html" %}

{% block announce %}
This lesson is under active development and considered pre-alpha.
{% endblock %}

{% block site_meta %}
{{ super() }}
<script type="application/ld+json">
  {
    "@context": "https://schema.org/",
    "@type": "LearningResource",
    "@id": "https://elixir-europe-training.github.io/ELIXIR-TrP-LessonTemplate-MkDocs/",
    "http://purl.org/dc/terms/conformsTo": {
        "@type": "CreativeWork",
        "@id": "https://bioschemas.org/profiles/TrainingMaterial/1.0-RELEASE"
    },
    "description": "Template for ELIXIR lessons",
    "keywords": "FAIR, OPEN, Bioinformatics, Teaching",
    "name": "ELIXIR Training Lesson template",
    "license": "https://creativecommons.org/licenses/by/4.0/",
    "educationLevel": "beginner",
    "competencyRequired": "none",
    "teaches": "nothing yet to teach"
}
</script>
{% endblock %}
```

You should probably use the `scripts` block rather than the `site_meta` one by the way.

**Reply by penyuan** [reply]
Aha, that worked. Thank you so much @pawamoy! Really appreciate your detailed support. ❤️

**Reply by pawamoy** [reply]
You're welcome :slightly_smiling_face: Thanks for coming back and selecting the answer :slightly_smiling_face:
